element-hq/element-x-android
# Feature/bma/fix fdroid notification

## Type of change


## Content


Fix issue on code running twice, and add more log regarding pusher registration.

This change also include a dialog to help user with no distributor installed. For UnifiedPush, this will act like this:

![NoDistributors](https://github.com/element-hq/element-x-android/assets/3940906/8856acb1-f631-4ddf-8eb4-6ebeeb9f0e78)

Also: if several push distributors are available (Firebase + UnifiedPush), the application will use Firebase by default (as before), but if not available, it will fallback to UnifiedPush is a UnifiedPush distributor is available on the system.

## Motivation and context


Help investigate #3031 

## Screenshots / GIFs



## Tests



- Install Ntfy
- Use F-Droid variant
- Connect to a Matrix account
- Check that the notification troubleshoot does not report any error
- The application is receiving push and notification are displayed.

## Tested devices

- OS version(s):

## Checklist
